macos/snd_.ksy

```diff
@@ -0,0 +1,201 @@
+enums:
+  data_type:
+    0x01: square_wave_synth
+    0x03: wave_table_synth
+    0x05: sampled_synth
+  cmd_type:
+    00: null_cmd
+    03: quiet_cmd
+    04: flush_cmd
+    10: wait_cmd
+    11: pause_cmd
+    12: resume_cmd
+    13: call_back_cmd
+    14: sync_cmd
+    15: empty_cmd
+    40: freq_duration_cmd
+    41: rest_cmd
+    42: freq_cmd
+    43: amp_cmd
+    44: timbre_cmd
+    60: wave_table_cmd
+    61: phase_cmd
+    80: sound_cmd
+    81: buffer_cmd
+    82: rate_cmd
+  sound_header_type:
+    0x00: standard
+    0xFF: extended
+    0xFE: compressed
+seq:
+  - id: format
+    type: u2
+  - id: number_of_data_formats
+    type: u2
+    if: format==1
+  - id: data_format
+    type: data_format
+    repeat: expr
+    repeat-expr: number_of_data_formats
+    if: format==1
+  - id: reference_count
+    type: u2
+    if: format==2
+  - id: number_of_sound_commands
+    type: u2
+  - id: sound_command
+    type: sound_command
+    repeat: expr
+    repeat-expr: number_of_sound_commands
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    instances:
+      cmd:
+        value: raw_cmd&0x7FFF
+        enum: cmd_type
+      is_data_offset:
+        value: raw_cmd&0x8000
+      sound_header_type:
+        pos: param2 + 20
+        type: u1
+        enum: sound_header_type
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
+      sound_header:
+        pos: param2
+        size-eos: true
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
+        type:
+          switch-on: sound_header_type
+          cases:
+            sound_header_type::standard: standard_sound_header
+            sound_header_type::extended: extended_sound_header
+            sound_header_type::compressed: compressed_sound_header
+    seq:
+      - id: raw_cmd
+        type: u2
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+  standard_sound_header:
+    seq:
+      - id: sample_ptr
+        type: u4
+      - id: lenght
+        type: u4
+      - id: sample_rate
+        type: u2
+      - id: sample_rate_frac
+        type: u2
+      - id: loop_start
+        type: u4
+      - id: loop_end
+        type: u4
+      - id: encode
+        type: u1
+        enum: sound_header_type
+      - id: base_frequency
+        type: u1
+      - id: sample_area
+        size: lenght
+  extended_sound_header:
+    seq:
+      - id: sample_ptr
+        type: u4
+      - id: num_channels
+        type: u4
+      - id: sample_rate
+        type: u2
+      - id: sample_rate_frac
+        type: u2
+      - id: loop_start
+        type: u4
+      - id: loop_end
+        type: u4
+      - id: encode
+        type: u1
+        enum: sound_header_type
+      - id: base_frequency
+        type: u1
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        #this is a float80
```

Comment:
Relevant - #210 adds specs for floating-point types that KS doesn't support natively, including 80-bit floats. That PR hasn't been merged yet though, so it can't be used here at the moment.